internal/tofu/transform_module_variable.go

```diff
@@ -133,3 +136,31 @@ func (t *ModuleVariableTransformer) transformSingle(g *Graph, parent, c *configs
 
 	return nil
 }
+
+// moduleImportedFromRemote is traversing upwards from the given module call to the root module and is looking for any
+// module on the path for which configs.Module.EntersNewPackage=true.
+// This is needed to know if a variable is referenced from a module imported from a remote source or from a local one.
+func moduleImportedFromRemote(c *configs.Config, call *configs.ModuleCall) bool {
```

Comment:
This appears to be a duplicate function from evaluate.go:isCallFromRemote

Could we move this common functionality to the configs package as it's only operating on objects that live there?

`func (c *configs.Config) IsModuleCallFromRemoteModule(call *configs.ModuleCall) bool`

Reply:
There is one important difference: one is using configs.ModuleCall and the other is using addrs.ModuleCall.
If you know a way to generate a addrs.ModuleCall from a configs.ModuleCall, I would gladly get rid of one of those.

Reply:
Ah, thanks.  Let me take a look

Reply:
I think that other than the `if _, ok := c.SourceAddr.(addrs.ModuleSourceRemote); ok {` check, the rest of the code is identical.  We could just pass the calledModuleName in to the common function instead of a more complex string.

Reply:
Funny how I missed that opportunity. Thanks.
Changed in 9782eb2495d09b5869fe8ede4e57447e4aceeb22